count/Feature/TopicsFeature.swift

```diff
@@ -165,16 +199,31 @@ struct TopicsView: View {
                     }
                 }
             }
-        }
-        .fullScreenCover(
-            store: store.scope(state: \.$quiz, action: { .quiz($0) })
-        ) { store in
-            ListeningQuizNavigationView(store: store)
-        }
-        .sheet(
-            store: store.scope(state: \.$about, action: { .about($0) })
-        ) { store in
-            AboutView(store: store)
+            .fullScreenCover(
+                store: store.scope(state: \.$destination, action: { .destination($0) }),
+                state: /TopicsFeature.Destination.State.quiz,
+                action: TopicsFeature.Destination.Action.quiz
+            ) { store in
+                ListeningQuizNavigationView(store: store)
+            }
+            .sheet(
+                store: store.scope(state: \.$destination, action: { .destination($0) }),
+                state: /TopicsFeature.Destination.State.about,
+                action: TopicsFeature.Destination.Action.about
+            ) { store in
+                AboutView(store: store)
+            }
+            .sheet(
+                store: store.scope(state: \.$destination, action: { .destination($0) }),
+                state: /TopicsFeature.Destination.State.preSettings,
+                action: TopicsFeature.Destination.Action.preSettings
+            ) { store in
+                PreSettingsView(store: store)
+                    .presentationDragIndicator(.visible)
+                    .presentationDetents([.fraction(0.1), .medium, .large])
+                    .presentationBackgroundInteraction(.enabled)
+                    .interactiveDismissDisabled(true)
+            }
```

Comment:
I noticed a bug (documented [on Mastondon](https://hachyderm.io/@twocentstudios/111363837645828560)) where SwiftUI presents this sheet as a `fullScreenCover` with no way to close it, but can't reproduce it.

At the time I saw the bug, the presentation modifiers were outside the `ViewStore` scope. I moved them inside (which I think is the correct placement) so hopefully that fixes it. There's not really much I can do to workaround it besides randomly placing delays, introspecting UIKit, or completely rewriting to not use `sheet` at all.